CRM/Contribute/BAO/Contribution.php

```diff
@@ -946,19 +952,11 @@ public static function recordPaymentActivity($contributionId, $participantId, $t
    * @return int
    */
   public static function getToFinancialAccount($contribution, $params) {
-    $contributionStatuses = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
-    CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
-    $pendingStatus = [
-      array_search('Pending', $contributionStatuses),
-      array_search('In Progress', $contributionStatuses),
-    ];
-    if (in_array(CRM_Utils_Array::value('contribution_status_id', $contribution), $pendingStatus)) {
-      return CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contribution['financial_type_id'], 'Accounts Receivable Account is');
```

Comment:
This caused a test to fail & on review I think it would be always incorrect when adding a payment - the only way to reach this line of code

Reply:
There is a CiviContribute Component setting that determines if every contribution starts with an Accounts Receivable entry. If a payment is being created at time the contribution is being created then it will be subsequent to this Pay Later initial transaction. I think this line is in error in not checking the setting before deciding what to record first.

Reply:
@JoeMurray that's a bit cryptic - what do you think the removed lines did? From my testing payments against 'Pending' or the unused 'In Progress' wound up having a from and to account of Accounts Receivable with this line in there

Reply:
@eileenmcnaughton this line AFAIK decides to_financial_account_id when a contribution is created or updated. 

to_financial_account_id should be 
1. AR if contribution status is pending (dont recall about In Progress)
2. Else it should grab from payment processor FA if payment is dont through payment processor
3 Else it should grab from payment instrument if payment is not from payment processor

https://wiki.civicrm.org/confluence/pages/viewpage.action?pageId=390955013#CiviAccountsDataFlow-Onlinecompletedcontribution

Reply:
@pradpnayak it's only called from functions that create payments so scenario  1 should be void  - which is what is removed

Reply:
@eileenmcnaughton than it should be If payment-processor else payment-instrument

Reply:
@pradpnayak it does an early return ```if (!empty($params['payment_processor'])) {``` so it doesn't need the 'else'